Introducing new Cluster Group Affinity Rule commands (#3570)
MicrosoftDocs/windows-powershell-docs: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md

---
description: Remove-ClusterGroupFromAffinityRule
external help file: ClusterAffinityRule.cdxml-help.xml
Module Name: FailoverClusters
ms.date: 07/27/2023
online version: https://learn.microsoft.com/powershell/module/failoverclusters/remove-clustergroupfromaffinityrule?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
schema: 2.0.0
title: Remove-ClusterGroupFromAffinityRule
---

# Remove-ClusterGroupFromAffinityRule

## SYNOPSIS
Removes a cluster group from an affinity rule.

## SYNTAX

### Query (cdxml) (Default)

```
Remove-ClusterGroupFromAffinityRule [[-Name] <String[]>] [-Groups] <String[]>
 [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
```

### InputObject (cdxml)

```
Remove-ClusterGroupFromAffinityRule -InputObject <CimInstance[]> [-Groups] <String[]>
 [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
```

## DESCRIPTION

Removes a cluster group from an affinity rule. This doesn't remove the rule itself.

## EXAMPLES

> ### Example 1

Reviewer: Missing descriptive name after the example number.

```powershell
Remove-ClusterGroupFromAffinityRule -Name MuRyle -Groups MyGroup -Cluster MyCluster
```

> This example removes the cluster group MyGroup from the affinity rule named MyRule.

Reviewer: ```suggestion
This example removes the cluster group `MyGroup` from the affinity rule named `MyRule`.
```

## PARAMETERS

### -AsJob

Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
complete.

The cmdlet immediately returns an object that represents the job and then displays the command
prompt. You can continue to work in the session while the job completes. To manage the job, use the
`*-Job` cmdlets. To get the job results, use the
[Receive-Job](https://go.microsoft.com/fwlink/?LinkID=113372) cmdlet.

> For more information about Windows PowerShell background jobs, see

Reviewer: ```suggestion
For more information about PowerShell background jobs, see
```

[about_Jobs](https://go.microsoft.com/fwlink/?LinkID=113251).

```yaml
Type: SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -CimSession

Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
object, such as the output of a [New-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227967)
or [Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
current session on the local computer.

```yaml
Type: CimSession[]
Parameter Sets: (All)
Aliases: Session

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Groups

> The groups to be removed from the affinity rule.

Reviewer: ```suggestion
The groups to remove from the affinity rule.
```

```yaml
Type: String[]
Parameter Sets: (All)
Aliases:

Required: True
Position: 1
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -InputObject

> Specifies the input object that is used in a pipeline command.

Reviewer: Can this be more specific to the cmdlet?

```yaml
Type: CimInstance[]
Parameter Sets: InputObject (cdxml)
Aliases:

Required: True
Position: Named
Default value: None
Accept pipeline input: True (ByValue)
Accept wildcard characters: False
```

### -Name

> The name of the affinity rule to remove groups from 

Reviewer: ```suggestion
The name of the affinity rule to remove groups from.
```

```yaml
Type: String[]
Parameter Sets: Query (cdxml)
Aliases: Set

Required: False
Position: 0
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -PassThru

> Returns an object representing the item with which you are working. By default, this cmdlet doesn't

Reviewer: Can this be more descriptive of the output object and include its type?

generate any output.

```yaml
Type: SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -ThrottleLimit

Specifies the maximum number of concurrent operations that can be established to run the cmdlet. If

> this parameter is omitted or a value of `0` is entered, then Windows PowerShell&reg; calculates an

Reviewer: ```suggestion
this parameter is omitted or a value of `0` is entered, then PowerShell calculates an
```

optimum throttle limit for the cmdlet based on the number of CIM cmdlets that are running on the
computer. The throttle limit applies only to the current cmdlet, not to the session or to the
computer.

```yaml
Type: Int32
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### CommonParameters

This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable,
-InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose,
-WarningAction, and -WarningVariable. For more information, see
[about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).

## INPUTS

### System.String[]

### Microsoft.Management.Infrastructure.CimInstance[]

## OUTPUTS

### Microsoft.Management.Infrastructure.CimInstance

> ### Microsoft.Management.Infrastructure.CimInstance#root/MSCLUSTER/MSCluster_AffinityRule

Reviewer: Can we add descriptions for these types - when and how they're used or returned?
